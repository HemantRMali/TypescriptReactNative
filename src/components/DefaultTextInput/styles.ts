import {StyleSheet} from 'react-native';
export default StyleSheet.create({
  textInputContainer: {marginBottom: 30},
});
