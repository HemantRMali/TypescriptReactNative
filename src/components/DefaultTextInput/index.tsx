import React from 'react';
import {FloatingLabelInput} from 'react-native-floating-label-input';
import styles from './styles';

interface DefaultTextInputProps {
  isSecure?: boolean;
  label: string;
  value: string;
  testId?: string;
  onChangeText(text: string): void;
  placeholder?: string;
}

const DefaultTextInput: React.FC<DefaultTextInputProps> = (props) => {
  const onChangeText = (text: string) => {
    props.onChangeText(text);
  };
  return (
    <FloatingLabelInput
      testID={props.testId}
      containerStyles={styles.textInputContainer}
      label={props.label}
      placeholder={props.placeholder}
      value={props.value}
      secureTextEntry={props.isSecure}
      onChangeText={(text: string) => {
        onChangeText(text);
      }}
    />
  );
};

DefaultTextInput.defaultProps = {
  isSecure: false,
};
export default DefaultTextInput;
