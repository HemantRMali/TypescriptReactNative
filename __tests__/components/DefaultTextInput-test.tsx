// __tests__/DefaultTextInput-test.js

import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';

import DefaultTextInput from '../../src/components/DefaultTextInput';
describe('DefaultTextInput', () => {
  const onChangeTextMock = jest.fn();

  it('receives some text', () => {
    const {getByTestId, toJSON} = render(
      <DefaultTextInput
        testId="userInput"
        label="Test Label"
        value="Test Value"
        onChangeText={onChangeTextMock}
      />,
    );

    fireEvent.changeText(getByTestId('userInput'), 'Hemant Mali');
    expect(toJSON()).toMatchSnapshot();
  });
});
