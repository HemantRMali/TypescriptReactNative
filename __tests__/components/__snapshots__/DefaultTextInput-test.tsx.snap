// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultTextInput receives some text 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "backgroundColor": "#00000000",
      "borderColor": "#49658c",
      "borderRadius": 12,
      "borderWidth": 2,
      "color": "#49658c",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 30,
      "paddingBottom": 10,
      "paddingHorizontal": 11,
      "paddingTop": 10,
    }
  }
>
  <Text
    onPress={[Function]}
    style={
      Object {
        "color": "#49658c",
        "fontSize": 10,
        "left": 0,
        "paddingLeft": 10,
        "position": "absolute",
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": 14,
          },
        ],
        "zIndex": 3,
      }
    }
  >
    Test Label
  </Text>
  <TextInput
    allowFontScaling={true}
    containerStyles={
      Object {
        "marginBottom": 30,
      }
    }
    label="Test Label"
    onBlur={[Function]}
    onChangeText={[Function]}
    onFocus={[Function]}
    onSelectionChange={[Function]}
    onSubmitEditing={[Function]}
    rejectResponderTermination={true}
    secureTextEntry={false}
    style={
      Object {
        "color": "#49658c",
        "flex": 1,
        "minHeight": 28,
        "paddingVertical": 0,
        "zIndex": 10,
      }
    }
    testID="userInput"
    underlineColorAndroid="transparent"
    value="Test Value"
  />
  <View />
</View>
`;
